sale_order_line_packaging_qty/models/sale_order_line.py

```diff
@@ -86,5 +87,69 @@ def _onchange_product_uom_qty(self):
         # TODO Drop once https://github.com/odoo/odoo/pull/49150/ is merged
         res = super()._onchange_product_uom_qty()
         if not res:
-            res = self._check_package()
+            res = self._check_package() or self._check_qty_is_pack_multiple()
         return res
+
+    @api.onchange("product_id")
+    def product_id_change(self):
+        res = super().product_id_change()
+        if self.product_id.only_sell_by_pack:
+            self.product_uom_qty = min(self.product_id.packaging_ids.mapped("qty"))
+        return res
+
+    def _check_qty_is_pack_multiple(self):
+        """ Check only for product with only_sell_by_pack
+        """
+        # self.product_uom_qty != 1 cause is the defaut value
+        # and we dont want to have this warning when we had the product
+        if self.product_id.only_sell_by_pack:
+            if not self._is_pack_multiple():
+                warning_mess = {
+                    "title": _("Product quantity can not be packed"),
+                    "message": _(
+                        "For the product {prod}\n"
+                        "The {qty} is not the multiple of any pack.\n"
+                        "Please add a package".format(
+                            prod=self.product_id.name, qty=self.product_uom_qty
+                        )
+                    ),
+                }
+                return {"warning": warning_mess}
+        return {}
+
+    def _is_pack_multiple(self):
+        is_pack_multiple = False
```

Comment:
This function is really closed to `def _which_pack_multiple(self, qty)` on `product.product` 
I wonder if you could just do something like this
```suggestion
        return bool(self.product_id._which_pack_multiple(self.product_uom_qty))
```